reconcile/integrations_manager.py

```diff
@@ -22,10 +24,70 @@
 QONTRACT_INTEGRATION_VERSION = make_semver(0, 1, 0)
 
 
+@dataclass
+class IntegrationShardManager:
+
+    aws_accounts: list[dict[str, Any]]
+    integration_runtime_meta: dict[str, IntegrationMeta]
+
+    def build_integration_shards(
+        self, integration: str, spec: Mapping[str, Any]
+    ) -> list[dict[str, Any]]:
+        extra_args = spec["extraArgs"] or ""
+        sharding_strategy = spec.get("shardingStrategy") or "static"
+        if sharding_strategy == "per-aws-account":
```

Comment:
To avoid the if/elif/else, reduce the size of the function (scrolling hell) to have smaller method and foreseeing  future sharding strategies, this is screaming at me the use of a [dispatch table](https://betterprogramming.pub/dispatch-tables-in-python-d37bcc443b0b)